src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -101,6 +109,7 @@ public KubernetesCloud(String name, List<? extends PodTemplate> templates, Strin
 
         this.serverUrl = serverUrl;
         this.namespace = namespace;
+
```

Comment:
IIUC this gets called when you save the plugin configuration. When the configuration is already saved then readResolve and methods marked with DataBoundSetter are called